model-client/src/commonMain/kotlin/org/modelix/model/client2/ReplicatedModel.kt

```diff
@@ -32,20 +32,23 @@ class ReplicatedModel(
     private val scope = providedScope ?: CoroutineScope(Dispatchers.Default)
     private var state = State.New
     private lateinit var localModel: LocalModel
-    private val remoteVersion = RemoteVersion(client, branchRef)
+    private lateinit var remoteVersion: RemoteVersion
     private var pollingJob: Job? = null
 
     fun getBranch(): IBranch {
         if (state != State.Started) throw IllegalStateException("state is $state")
         return localModel.otBranch
     }
 
-    suspend fun start(): IBranch {
+    suspend fun start(providedInitialVersion: CLVersion? = null, onBeforeSync: (IBranch) -> Unit = {}): IBranch {
```

Comment:
* Wouldn't it be more consistent to also provide `providedInitialVersion` via the constructor, similar to the branch reference?
* Why don't we use the same name as in the `RemoteVersion` object: `lastKnownRemoteVersion`?

Reply:
Moving `providedInitialVersion` to the constructor also allows initializing `localModel` earlier and then the `onBeforeSync` callback isn't necessary.

Reply:
We need the `onBeforeSync` call, because we would like to subscribe to changes to the IBranch as soon as it is fetched from the server (via the LocalModel). Is there any other way to accomplish this?

Reply:
fixed in [86dd24c](https://github.com/modelix/modelix.core/pull/725/commits/86dd24c1183e2cf0213864f2d56d0d134077568f), except that the `onBeforeSync` has to stay, because we would like to subscribe to changes on the IBranch, before we fetch the latest version from the server.

Reply:
@slisson fine with that now?

Reply:
I you provide a `lastKnownRemoteVersion` to the constructor then you can initialize `localModel` already in the constructor instead of in `start`. You can then call `getBranch()` and register your listener before calling `start()`, which means the `onBeforeSync` callback isn't necessary anymore.

Reply:
I'll make this change then.

Reply:
fixed in [9b0ea3f](https://github.com/modelix/modelix.core/pull/725/commits/9b0ea3fa558315b5a31846a9057adb2e484365ba)

Reply:
@slisson is the fix in [9b0ea3f](https://github.com/modelix/modelix.core/pull/725/commits/9b0ea3fa558315b5a31846a9057adb2e484365ba) fine for you?

Reply:
I changed the commit a bit. Now it can be merged.